### PR TITLE
Feature last page repeat problem

### PR DESCRIPTION
**Pull request by dkhasham**
Sende eine leere Liste von Events, wenn letzte Paging-Seite überschritten ist
Send an empty list of events if the last Page is passed by
Related Ticket: https://synjira.atlassian.net/browse/CELDEV-231

### CODE REVIEW

**Comment by fpichler** [comment]
@msladek can you please do this code review, because the classes involved have been written by you. :-)

**Comment by msladek** [comment]
@fpichler sure

**Comment by fpichler** [comment]
please add a CELDEV JIRA issue for this change.

File: component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java [hunk 1]
```diff
@@ -157,9 +185,23 @@ private NavigationDetails getLastNavDetails(ICalendar cal, int offset
...
+    return getCalendarNavigation(calDocRef, navDetails, nb, query, false);
+  }
+  
   public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
-      NavigationDetails navDetails, int nb, DateEventSearchQuery query
-      ) throws LuceneSearchException {
+      NavigationDetails navDetails, int nb, DateEventSearchQuery query, 
+      boolean isSendingEmptyPage) throws LuceneSearchException {
+    System.out.println("<<<<<<<<<<<<<<<<<<<<<<<<<< CalendarNavigationFactory getCalendarNavigation offset: [" 
+      + navDetails.getOffset() + "]");
```
Comment (msladek) [inline component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java]: Use Logger or remove System.out call

File: component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java [hunk 2]
```diff
@@ -189,12 +231,39 @@ public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
...
-        getPrevNavDetails(calDocRef, navDetails, nb, query, calResult, calArchiveResult),
-        getNextNavDetails(calDocRef, navDetails, nb, query, calResult));
-    LOGGER.debug("getCalendarNavigation: return '" + calendarNavigation + "' for cal '"
-        + calDocRef + "', navDetails '" + navDetails + "' and query '" + query + "'");
+        prevNavDetails, nextNavDetails);
+    System.out.println("<<<<<<<<<<<<<<<<<<<<<<<<<< CalendarNavigationFactory getCalendarNavigation countAfter: ["
+        + calendarNavigation.getCountAfter().getCount()
+        + "] offset: [" + navDetails.getOffset()
+        + "] nb: [" + nb + "] actualOffset: [" + actualOffset + "]");
+    System.out.println();
```
Comment (msladek) [inline component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java]: Use Logger or remove System.out call

File: component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java [hunk 3]
```diff
@@ -58,6 +68,24 @@ public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
...
+      try {
+        calendarNavigation = new CalendarNavigation(
+            new UncertainCount(0, false),
+            new UncertainCount(0, false),
+            new UncertainCount(0, false), 
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0));
```
Comment (msladek) [inline component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java]: Extract to own method for reusability and code readability

File: component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java [hunk 4]
```diff
@@ -189,12 +231,39 @@ public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
...
```
Comment (msladek) [inline component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java]: Extract to own method for reusability and code readability

File: component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java [hunk 5]
```diff
@@ -58,6 +68,24 @@ public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
...
+      } catch (NavigationDetailException exc) {
+        LOGGER.error("Exception getting calNav for cal [" + calDocRef + "], eventDate ["
+            + navDetails.getStartDate() + "], " + "offset [" + navDetails.getOffset()
+            + "], nb ["+ nb +"]", exc);
+      }
```
Comment (msladek) [inline component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java]: Extract to own method for reusability and code readability.

Also, NavigationDetailException should never happen in this case -> perhaps rethrow as a RuntimeException?

File: component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java [hunk 6]
```diff
@@ -189,12 +231,39 @@ public CalendarNavigation getCalendarNavigation(DocumentReference calDocRef,
...
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0),
+            NavigationDetails.create(new Date(Long.MAX_VALUE), 0));
+      } catch (NavigationDetailException exc) {
+        LOGGER.error("Exception getting calNav for cal [" + calDocRef + "], eventDate ["
+            + navDetails.getStartDate() + "], " + "offset [" + navDetails.getOffset()
+            + "], nb ["+ nb +"], query ["+ query +"]", exc);
+      }
```
Comment (msladek) [inline component/src/main/java/com/celements/calendar/navigation/factories/CalendarNavigationFactory.java]: Extract to own method for reusability and code readability.

Also, NavigationDetailException should never happen in this case -> perhaps rethrow as a RuntimeException?

**Comment by msladek** [comment]
A few things to consider, looks good otherwise.

**Comment by dkhasham** [comment]
all changes implemented, please merge if everything is ok

**Pull request merged**